### ISSUE TITLE
[Bug report] Create Date sql query not working properly

### ISSUE THREAD

**Issue by priyanshu-simformsolutions** [fluttercandies/flutter_photo_manager#1242]
### Version

3.6.2

### Platforms

Android

### Device Model

All

### flutter info

```sh
[✓] Flutter (Channel stable, 3.22.2, on macOS 14.3.1 23D60 darwin-arm64,
    locale en-IN)
    • Flutter version 3.22.2 on channel stable at
      /Users/{My_Path}/fvm/versions/3.22.2
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 761747bfc5 (6 months ago), 2024-06-05 22:15:13 +0200
    • Engine revision edd8546116
    • Dart version 3.4.3
    • DevTools version 2.34.3

[✓] Android toolchain - develop for Android devices (Android SDK version
    34.0.0)
    • Android SDK at /Users/{My_Path}/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      17.0.10+0-17.0.10b1087.21-11572160)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15E204a
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      17.0.10+0-17.0.10b1087.21-11572160)

[✓] Connected device (7 available)
    • Pixel 6a (mobile)               • **********            •
      android-arm64  • Android 14 (API 34)

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

### How to reproduce?

When filtering images through CustomColumns.base.createDate on IOS it seems to work as expected,
while in Android i am getting image even from 2-3 years old createDate

Steps i have tried:
- I have even tried with same images on both platform, still same issue happens.
- I have read all meta data of an image it has same date in all fields createDate(date_added), date_taken, modified_date etc... ex(image date is of 2022) and still seems to able to pass this filter even in 2024 (i.e.: According to above filter only last one year image should be able to be fetched) 
- I have also tried same thing AdvancedCustomFilter, same issue occurs.

There is some issue with create date filtering on Android side query i guess.

### Example code (optional)

```Dart
PMFilter? get recentFilterOption {
    final prevYearSqlDateTimeStamp = CustomColumns.utils.convertDateTimeToSql(
      DateTime.now().previousYear, // Used below extension method
      isSeconds: Platform.isAndroid,
    );

    return CustomFilter.sql(
      where: '$commonFilterQuery AND '
          '${CustomColumns.base.createDate} >= $prevYearSqlDateTimeStamp',
      orderBy: [OrderByItem.desc(CustomColumns.base.createDate)],
    );
  }

/// Returns [DateTime] with previous year
  DateTime get previousYear => DateTime(
      year - 1, month, day, hour, minute, second, millisecond, microsecond);
```

**Comment by AlexV525** [comment]
Seems no issues on my device.

![image](https://github.com/user-attachments/assets/8a276ffe-868f-4337-b552-46723c7540f2)
![image](https://github.com/user-attachments/assets/ae52a0f6-19e5-4b1a-830a-3e1b57dfc572)
![image](https://github.com/user-attachments/assets/ed7646bd-423a-4d3e-ba1d-768d9fe843ea)

**Comment by priyanshu-simformsolutions** [comment]
As you can refer this recording
i have 6 photos 3 of Nov 24, and other 3 of Oct 22

now when applying filter of date_added grater then Dec 2023,
and i still get all the photos 

Note: I have printed CreateDate from AssetEntity
This issue only occurs in Android.

https://github.com/user-attachments/assets/9b4fbf10-a7aa-41c7-9b9f-469c85e8efb4

**Comment by AlexV525** [comment]
Was the query not working as always regardless of the date? Or was the feature not working on this single phone and provide it's model?

**Comment by AlexV525** [comment]
It sounds like a device-specific problem.

**Comment by priyanshu-simformsolutions** [comment]
Hey Alex, by debugging more on this i was able to reproduce it only happens to picture-specific and only in android(any android device), 
but it is not rare as i can see many of my images to have this issue (i have almost 400 such images, all from different sources), 
i am sharing two images for example in this [link](https://photos.app.goo.gl/Jpcz3eyv1oJbUWPk7),
This both images are from Dec 05, 2022. (If you want more such images for testing i can send it to you)

Before filter
<img width="737" alt="Screenshot 2024-12-12 at 10 25 53 PM" src="https://github.com/user-attachments/assets/0c463534-4b0e-405d-af62-4c94f9cb3441" />

Filter
<img width="1064" alt="Screenshot 2024-12-12 at 10 25 30 PM" src="https://github.com/user-attachments/assets/dbfcb7c8-4a92-4b13-a298-64bbc9b9b9f3" />

After Filter
<img width="742" alt="Screenshot 2024-12-12 at 10 26 04 PM" src="https://github.com/user-attachments/assets/f920abf6-1200-4b3b-b7c0-72bfdac34602" />

**Comment by AlexV525** [comment]
How do you read the creation date in the grid?

**Comment by priyanshu-simformsolutions** [comment]
To read the creation date in the grid, I use AssetEntity.createDateTime.

In the ImageItemWidget, I modified the _buildImageWidget() method as follows:

Note: I use an extension method getReadableDateFormat() to format the DateTime as needed.

`Widget _buildImageWidget(
    BuildContext context,
    AssetEntity entity,
    ThumbnailOption option,
  ) {
    return Stack(
      children: <Widget>[
        Positioned.fill(
          child: AssetEntityImage(
            entity,
            isOriginal: false,
            thumbnailSize: option.size,
            thumbnailFormat: option.format,
            fit: BoxFit.cover,
          ),
        ),
        PositionedDirectional(
          bottom: 4,
          start: 0,
          end: 0,
          child: Row(
            children: [
              if (entity.isFavorite)
                Container(
                  padding: const EdgeInsets.all(4),
                  decoration: BoxDecoration(
                    color: Theme.of(context).cardColor,
                    shape: BoxShape.circle,
                  ),
                  child: const Icon(
                    Icons.favorite,
                    color: Colors.redAccent,
                    size: 16,
                  ),
                ),
              Expanded(
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.end,
                  children: [
                    if (entity.isLivePhoto)
                      Container(
                        margin: const EdgeInsetsDirectional.only(end: 4),
                        padding: const EdgeInsets.symmetric(
                          horizontal: 3,
                          vertical: 1,
                        ),
                        decoration: BoxDecoration(
                          borderRadius: const BorderRadius.all(
                            Radius.circular(4),
                          ),
                          color: Theme.of(context).cardColor,
                        ),
                        child: const Text(
                          'LIVE',
                          style: TextStyle(
                            color: Colors.black,
                            fontSize: 10,
                            fontWeight: FontWeight.bold,
                            height: 1,
                          ),
                        ),
                      ),
                    Icon(
                      () {
                        switch (entity.type) {
                          case AssetType.other:
                            return Icons.abc;
                          case AssetType.image:
                            return Icons.image;
                          case AssetType.video:
                            return Icons.video_file;
                          case AssetType.audio:
                            return Icons.audiotrack;
                        }
                      }(),
                      color: Colors.white,
                      size: 16,
                    ),
                  ],
                ),
              ),
            ],
          ),
        ),
        Center(
          child: Text(
            entity.createDateTime.getReadableDateFormat(),
            style: const TextStyle(color: Colors.white),
          ),
        ),
      ],
    );
  }
`

**Comment by priyanshu-simformsolutions** [comment]
Hi @AlexV525 ,

I hope you're doing well! I just wanted to check in—were you able to reproduce the issue? Please feel free to reach out if you need anything else.

**Comment by AlexV525** [comment]
No, I'm not actively investigating this. We process issues mainly in our spare time so it may take a while.

**Comment by oncelot** [comment]
I also have the same issue. For me, it only detects photos from 2024, but I modified the filter with updateTimeCond, and it also finds photos prior to 2024, for example, from 2023 and 2021. However, I can't find the photos from 2022. Very strange.

My old photos come from the SD card and were taken with other phones. Do you think it's possible that Android has its own internal database? Maybe it's possible to rescan the memory?

 FilterOptionGroup options = FilterOptionGroup(
    updateTimeCond: 
  DateTimeCond(
    min: DateTime(targetDate.year-i, targetDate.month, targetDate.day),
     max: DateTime(targetDate.year-i, targetDate.month, targetDate.day + 1)
     ),);
 
  final List<AssetPathEntity> albums = await PhotoManager.getAssetPathList( 
    hasAll: true,
    onlyAll: false,
    type: RequestType.image,
    filterOption: options,
  );
